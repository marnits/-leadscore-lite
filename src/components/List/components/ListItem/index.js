import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles.module.scss';
import columnType from '../../../../utils/propTypes/column';

const ListItem = ({ columns, children }) => {
  const renderColumn = (child, index) => {
    if (!columns[index]) {
      return null;
    }

    const { align = 'left', size = 1 } = columns[index];

    return (
      <div
        className={`
          ${styles.column}
          ${styles[`align-content-${align}`]}
          ${styles[`flex-${size}`]}
        `}
        key={columns[index].name}
      >
        {child}
      </div>
    );
  };

  return (
    <div className={styles.row}>
      {children.map(renderColumn)}
    </div>
  );
};

ListItem.defaultProps = {
  columns: [],
};

ListItem.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node),
  ]).isRequired,
  columns: PropTypes.arrayOf(columnType),
};

export default ListItem;
