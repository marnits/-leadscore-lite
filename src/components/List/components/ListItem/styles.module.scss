@import "../../../../utils/styles/colors.scss";

.row {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  min-height: 48px;
  max-height: 100%;
  border: 1px solid $border--color;
  align-items: center;
  outline: none;
  background-color: $white;
  border-bottom: 0;

  .column {
    display: flex;
    align-items: center;
    min-width: 0;
    padding: 8px 16px;

    &.align-content- {
      &left {
        text-align: left;
        justify-content: flex-start;
      }

      &center {
        text-align: center;
        justify-content: center;
      }

      &right {
        text-align: right;
        justify-content: flex-end;
      }
    }

    &.flex-1 { flex: 1 }
    &.flex-2 { flex: 2 }
    &.flex-3 { flex: 3 }
  }
}
