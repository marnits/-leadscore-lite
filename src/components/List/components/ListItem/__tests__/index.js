import React from 'react';
import renderer from 'react-test-renderer';
import ListItem from '../index';

const columns = [{
  name: 'Test',
}, {
  name: 'ABC',
}, {
  name: 'XYZ',
}];

describe('ListItem', () => {
  test('should render component properly', () => {
    const tree = renderer.create(
      <ListItem columns={columns}>
        <p>test</p>
        <span>test 2</span>
        <span>test 3</span>
      </ListItem>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('should render component properly', () => {
    const [first, ...rest] = columns;
    const tree = renderer.create(
      <ListItem columns={rest}>
        <p>{first.name}</p>
        <span>test 2</span>
        <span>test 3</span>
      </ListItem>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
