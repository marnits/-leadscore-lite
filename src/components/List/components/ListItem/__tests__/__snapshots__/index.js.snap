// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should render component properly 1`] = `
<div
  className="row"
>
  <div
    className="
          column
          align-content-left
          flex-1
        "
  >
    <p>
      test
    </p>
  </div>
  <div
    className="
          column
          align-content-left
          flex-1
        "
  >
    <span>
      test 2
    </span>
  </div>
  <div
    className="
          column
          align-content-left
          flex-1
        "
  >
    <span>
      test 3
    </span>
  </div>
</div>
`;

exports[`ListItem should render component properly 2`] = `
<div
  className="row"
>
  <div
    className="
          column
          align-content-left
          flex-1
        "
  >
    <p>
      Test
    </p>
  </div>
  <div
    className="
          column
          align-content-left
          flex-1
        "
  >
    <span>
      test 2
    </span>
  </div>
</div>
`;
